simplemma/dictionary_factory.py

```diff
@@ -0,0 +1,57 @@
+"""Parts related to dictonaries."""
+import lzma
+import logging
+import pickle
+
+from pathlib import Path
+from typing import Any, Dict, Optional, Tuple, Union
+
+from functools import lru_cache
+
+from .constants import LANGLIST
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _control_lang(lang: Any) -> Tuple[str]:
+    "Make sure the lang variable is a valid tuple."
+    # convert string
+    if isinstance(lang, str):
+        lang = (lang,)
+    if not isinstance(lang, tuple):
+        raise TypeError("lang argument must be a two-letter language code")
+    return lang  # type: ignore[return-value]
+
+
+def _load_dictionary_from_disk(langcode: str) -> Dict[str, str]:
+    filename = f"data/{langcode}.plzma"
+    filepath = str(Path(__file__).parent / filename)
+    with lzma.open(filepath, "rb") as filehandle:
+        pickled_dict = pickle.load(filehandle)
+        assert isinstance(pickled_dict, dict)
+        return pickled_dict
+
+
+class DictionaryFactory:
+    def __init__(self, cache_max_size: Optional[int] = 5):
```

Comment:
A power of two for cache size could be better, I suggest 8.
https://stackoverflow.com/questions/61536704/why-does-python-lru-cache-performs-best-when-maxsize-is-a-power-of-two

Reply:
Solved!

Reply:
Changed to 4 but changed now to 8